multigeiger/speaker.cpp

```diff
@@ -58,19 +59,20 @@ void IRAM_ATTR isr_audio() {
       playing_tick = true;
     }
   }
+
   if (isr_sequence) {
     volatile int *p = isr_sequence;
     frequency_mHz = *p++;
     volume = *p++;
     led = *p++;
     duration_ms = *p++;
-    isr_sequence = p;
+    isr_sequence = duration_ms == 0 ? NULL : p;
+  } else {
+    portEXIT_CRITICAL_ISR(&mux_audio);
+    return;  // nothing to do
   }
   portEXIT_CRITICAL_ISR(&mux_audio);
 
-  if (!isr_sequence)
```

Comment:
@ThomasWaldmann, here we accessed isr_sequence after exiting the &mux_audio. In the PR this is prevented. Will create a separate PR once the code compiles with IotWebConf 3.1.x.

Reply:
one could avoid having the portEXIT_CRITICAL_ISR(&mux_audio) duplicated to the else block by setting p = NULL there and afterwards just checking the value of p (local, instead of accessing global volatile isr_sequence).

smaller diff and easier mux ops.

Reply:
or rather:
```
  volatile int *p = isr_sequence;
  if (p) {
        ...
        isr_sequence = duration_ms == 0 ? NULL : p;  
  }
  portEXIT_CRITICAL_ISR(&mux_audio);

  if (!p)
    return;  // nothing to do
```

Reply:
Ok, adapted your proposition. Separate PR in preparation.

Reply:
thanks!